src/pages/organize/[orgId]/campaigns/[campId]/calendar/tasks/[taskId]/assignees.tsx

```diff
@@ -45,33 +50,84 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     };
 }, scaffoldOptions);
 
+enum QUERY_STATUS {
+    EDITABLE='editable',
+    PUBLISHED='published',
+    PUBLISHING='publishing',
+}
+
 const TaskAssigneesPage: PageWithLayout = () => {
     const { taskId, orgId } = useRouter().query;
     const taskQuery = useQuery(['task', orgId, taskId], getTask(orgId as string, taskId as string));
+    const assignmentsQuery = useQuery(['assignments', orgId, taskId], getTaskAssignments(
+        orgId as string, taskId as string,
+    ));
     const task = taskQuery?.data;
-
+    const assignments = assignmentsQuery?.data;
     const query = task?.target;
 
     const [dialogOpen, setDialogOpen] = useState(false);
 
     const handleDialogClose = () => setDialogOpen(false);
 
+    const taskStatus = task ? getTaskStatus(task) : undefined;
+
+    const getQueryStatus = () => {
+        let queryStatus = QUERY_STATUS.PUBLISHED;
+        if (taskStatus === TASK_STATUS.DRAFT || taskStatus === TASK_STATUS.SCHEDULED) {
+            queryStatus = QUERY_STATUS.EDITABLE;
+        }
+        // we don't want 'publishing' state to appear on page load while the data is being fetched
+        else if (assignments && !assignments.length) {
+            queryStatus = QUERY_STATUS.PUBLISHING;
+        }
+        return queryStatus;
+    };
+
+    const queryStatus = getQueryStatus();
+
+    const readOnly = queryStatus === QUERY_STATUS.PUBLISHING ||
+        queryStatus === QUERY_STATUS.PUBLISHED;
+
     return (
         <>
-            <Button color="primary" onClick={ () => setDialogOpen(true) } variant="contained">
-                Edit
-            </Button>
+            <Box p={ 2 }>
+                <Alert severity={
+                    queryStatus === QUERY_STATUS.EDITABLE ?
```

Comment:
I think the alert should be red if the Smart Search query has not been defined yet (i.e. if `filter_spec` is empty). The user _must_ define a Smart Search query, so not having one is an error.

Now that I see it, I also think that the "edit" link should be a button _in the error state_ because it's a required action. Do you think that sounds right?

Reply:
Yes, it sounds right, I will make a fourth state for the query and call it `UNDEFINED` that triggers if the filter_spec is empty